flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java

```diff
@@ -261,7 +261,7 @@ private Optional<String> validateSpecChange(FlinkDeployment deployment) {
                     && deployment.getStatus().getJobManagerDeploymentStatus()
                             != JobManagerDeploymentStatus.MISSING
                     && ReconciliationUtils.isUpgradeModeChangedToLastStateAndHADisabledPreviously(
-                            deployment)) {
+                            deployment, new Configuration())) {
```

Comment:
A concern for the refactor:
I do agree that introducing the defaultConfig in this validation as it is possible that some users may modify the default filnk-conf.yaml(e.g. adding HA config). But I am a little concerned that should we pass en empty config or using `FlinkUtils.loadConfiguration(EnvUtils.get(EnvUtils.ENV_FLINK_CONF_DIR))` ? So the Validator can work correctly if users do not specify HA config in Spec but add them in flink-conf.yaml.

Reply:
Good catch, The only reason I have this here because I am waiting for this PR to be completed by @SteNicholas : https://github.com/apache/flink-kubernetes-operator/pull/162

I didnt want to double implement the logic

Reply:
But I will add the proper config there anyways, as you suggested :)